frontend/src/__tests__/cypress/cypress/tests/e2e/pipelines/pipelines.cy.ts

```diff
@@ -0,0 +1,194 @@
+import {
+  createOpenShiftProject,
+  deleteOpenShiftProject,
+  applyOpenShiftYaml,
+} from '~/__tests__/cypress/cypress/utils/ocCommands';
+import { replacePlaceholdersInYaml } from '~/__tests__/cypress/cypress/utils/yaml_files';
+import { ADMIN_USER } from '~/__tests__/cypress/cypress/utils/e2eUsers';
+import { AWS_PIPELINES_BUCKET } from '~/__tests__/cypress/cypress/utils/s3Buckets';
+
+import { projectListPage, projectDetails } from '~/__tests__/cypress/cypress/pages/projects';
+import { pipelineImportModal } from '~/__tests__/cypress/cypress/pages/pipelines/pipelineImportModal';
+import { pipelinesGlobal } from '~/__tests__/cypress/cypress/pages/pipelines/pipelinesGlobal';
+import { pipelinesTable } from '~/__tests__/cypress/cypress/pages/pipelines/pipelinesTable';
+import { createRunPage } from '~/__tests__/cypress/cypress/pages/pipelines/createRunPage';
+import { pipelineRunDetails } from '~/__tests__/cypress/cypress/pages/pipelines/topology';
+
+const projectName = 'test-pipelines-prj';
+const dspaSecretName = 'dashboard-dspa-secret';
+const allAvailableProjectsText = 'All available projects';
+const testPipelineName = 'test-pipelines-pipeline';
+const testRunName = 'test-pipelines-run';
+
+describe('An admin user can import and run a pipeline', { testIsolation: false }, () => {
+  before(() => {
+    // Provision a Project
+    createOpenShiftProject(projectName).then((result) => {
+      expect(result.code).to.eq(
+        0,
+        `ERROR provisioning ${projectName} Project
+                                    stdout: ${result.stdout}
+                                    stderr: ${result.stderr}`,
+      );
+    });
+
+    // Create a pipeline compatible Data Connection
+    const dataConnectionReplacements = {
+      NAMESPACE: projectName,
+      AWS_ACCESS_KEY_ID: Buffer.from(AWS_PIPELINES_BUCKET.AWS_ACCESS_KEY_ID).toString('base64'),
+      AWS_DEFAULT_REGION: Buffer.from(AWS_PIPELINES_BUCKET.AWS_REGION).toString('base64'),
+      AWS_S3_BUCKET: Buffer.from(AWS_PIPELINES_BUCKET.BUCKET_NAME).toString('base64'),
+      AWS_S3_ENDPOINT: Buffer.from(AWS_PIPELINES_BUCKET.AWS_ENDPOINT).toString('base64'),
+      AWS_SECRET_ACCESS_KEY: Buffer.from(AWS_PIPELINES_BUCKET.AWS_SECRET_ACCESS_KEY).toString(
+        'base64',
+      ),
+    };
```

Comment:
We need types and utility functions for this instead of tests duplicating it.

Reply:
Fixed in 
28d8d14